cni/network/plugin/main.go

```diff
@@ -226,40 +227,58 @@ func main() {
 		panic("network plugin start fatal error")
 	}
 
+	defer func() {
+		netPlugin.Stop()
+
+		// release cni lock
+		if errUninit := netPlugin.Plugin.UninitializeKeyValueStore(false); errUninit != nil {
+			log.Errorf("Failed to uninitialize key-value store of network plugin, err:%v.\n", errUninit)
+		}
+
+		executionTimeMs := time.Since(startTime).Milliseconds()
+
+		if err != nil {
+			reportPluginError(reportManager, tb, err)
+			panic("network plugin execute fatal error")
+		}
+
+		// Report CNI successfully finished execution.
+		reflect.ValueOf(reportManager.Report).Elem().FieldByName("CniSucceeded").SetBool(true)
+		reflect.ValueOf(reportManager.Report).Elem().FieldByName("OperationDuration").SetInt(executionTimeMs)
+
+		if cniReport.ErrorMessage != "" || cniReport.EventMessage != "" {
+			if err = reportManager.SendReport(tb); err != nil {
+				log.Errorf("SendReport failed due to %v", err)
+			} else {
+				log.Printf("Sending report succeeded")
+			}
+		}
+	}()
+
 	// Check CNI_COMMAND value for logging purposes.
 	cniCmd := os.Getenv(cni.Cmd)
 	log.Printf("CNI_COMMAND environment variable set to %s", cniCmd)
 
+	// used to dump state
+	if cniCmd == cni.CmdState {
```

Comment:
Where are we acquiring the file lock?

Reply:
https://github.com/Azure/azure-container-networking/blob/a01039a3125cf2e849228c01136b2e74e15d72a2/network/manager.go#L384

lock is taken here, should be changed to RLock

Reply:
For now, it is okay for now as we dont want any other CNI instance to update this state while reconcile is going on